fem/bilininteg.cpp

```diff
@@ -2174,20 +2174,31 @@ void VectorDiffusionIntegrator::AssembleElementMatrix(
    const int dim = el.GetDim();
    const int dof = el.GetDof();
    const int sdim = Trans.GetSpaceDim();
+
+   // If vdim is not set, set it to the space dimension;
+   vdim = (vdim <= 0) ? sdim : vdim;
    const bool square = (dim == sdim);
-   double w;
 
-   elmat.SetSize(sdim * dof);
+   if (VQ)
+   {
+      vcoeff.SetSize(vdim);
```

Comment:
I see this the other way around. It should be the vector or matrix coefficient to set `vdim` on line 2179. The integrators should not be in charge of modifying the coefficients.

Reply:
Maybe this is a confusing variable name? `vcoeff` is actually a `Vector` and *not* a `VectorCoefficient`. It is simply a variable used to store the result returned by the `VectorCoefficient`, `VQ`. I am not modifying the coefficient, just setting the size of the result `Vector` or `DenseMatrix` so that there is enough space to store the result returned by the coefficient.